node/resolvers/catalog/category.ts

```diff
@@ -9,6 +9,14 @@ interface Category {
   children: Category[],
 }
 
+const flatten: (categories: Category[]) => Category[] = (
+  categories: Category[]
+) => {
+  return categories.reduce((acc: Category[], cat) => {
+    return [...acc, cat, ...cat.children, ...flatten(cat.children)]
```

Comment:
I might not get it at first, but `...cat.children, ...flatten(cat.children)` wouldn't push duplicates into the array?